### DISCUSSION TITLE
TPI3 - Duda eventos repetitivos

### DISCUSSION THREAD

**Discussion by AlejoGol** [dds-jv/foro#377] [Consultas generales] [answered]
Buenas, este tipo de eventos me estan dando problemas y queria consultar algunas cosas.
Entiendo por las interfaces propuestas que un Evento Simple guarda su inicio y su fin como 2 atributos de tipo LocalDateTime. Ahora bien, al instanciar un Evento Repetitivo ademas de usar todo lo que tenga el evento simple le tengo que pasar una periodicidad, una cantidad de veces dentro de esta periodicidad (o sea cada 2 dias por ejemplo) y una fecha limite en la cual debo parar de repetir el evento. Esto es correcto?
Si este es el caso, tiene sentido que al instanciar un Evento Repetitivo genere una lista con pares inicio y fin para cada uno de los dias que suceda el evento o para ciertos requerimientos (por ejemplo si cae entre dos fechas) tendria que hacer los calculos en el momento? Esto viene principalmente a que en el enunciado dice "En algún futuro, estas repeticiones podrían ser infinitas, por lo que no deseamos precalcular y precargar cada instancia del evento como un evento separado."

**Comment by gastonprieto** [comment, marked as answer]
Hola @AlejoGol 

> al instanciar un Evento Repetitivo ademas de usar todo lo que tenga el evento simple le tengo que pasar una periodicidad, una cantidad de veces dentro de esta periodicidad (o sea cada 2 dias por ejemplo) y una fecha limite en la cual debo parar de repetir el evento. Esto es correcto?

no estoy muy seguro si todo lo del evento simple, pero en gran medida si.

> Si este es el caso, tiene sentido que al instanciar un Evento Repetitivo genere una lista con pares inicio y fin para cada uno de los dias que suceda el evento

Esto no se puede hacer, es lo que te comenta mas abajo que no podes hacer (el texto que vos comentaste).
